### ISSUE TITLE
Add undeprecated versions of deprecated lemmas for _<_ in `Data.Integer.Properties`

### ISSUE THREAD

**Issue by MatthewDaggitt** [agda/agda-stdlib#1016]
See #752 

At a minimum includes:
```agda
m+1≤n⇒m<n : ∀ {m n} → sucℤ m ≤ n → m < n
m<n⇒m+1≤n : ∀ {m n} → m < n → sucℤ m ≤ n
m≤pred[n]⇒m<n : ∀ {m n} → m ≤ pred n → suc m ≤ n
m<n⇒m≤pred[n] : ∀ {m n} → suc m ≤ n → m ≤ pred n
```

**Comment by Blaisorblade** [comment]
s/#752/#756.

**Comment by Blaisorblade** [comment]
Also: now that @gallais agda-presburger's compiles against stdlib `master`, I think a good test would be to upgrade it to remove the deprecation warnings. I suspect most lemmas on `<′` need an undeprecated statement — `<′-irrelevant` has one already. And ideally the deprecation warnings should be updated.

**Comment by MatthewDaggitt** [comment]
I've just had a scan through and after #1017 it looks like every deprecated lemma has a non-deprecated counter-part. I'm therefore closing this issue. 

> Also: now that @gallais agda-presburger's compiles against stdlib master, I think a good test would be to upgrade it to remove the deprecation warnings.

Not sure who's in charge of maintaining this, but whoever it is they can take a shot at it now.

**Comment by Blaisorblade** [comment]
> I've just had a scan through and after #1017 it looks like every deprecated lemma has a non-deprecated counter-part.

I think there's a misunderstanding — that issue wasn't fixed after my PR.
Take `m≤pred[n]⇒m<′n : ∀ {m n} → m ≤ pred n → m <′ n`, which is mentioned already in the OP.
I'm asking to add `m≤pred[n]⇒suc[m]≤n : ∀ {m n} → m ≤ pred n → suc m ≤ n`, define `m≤pred[n]⇒m<′n = m≤pred[n]⇒suc[m]≤n`, and mention `m≤pred[n]⇒suc[m]≤n` in the deprecation message.

**Comment by MatthewDaggitt** [comment]
> I think there's a misunderstanding

Ah yes there has been, apologies. Thanks for the example, I think I'm now on the same page as you.

Hmm so presumably for every lemma of interest, it's just a case of taking the lemma in its standard form, and applying `m+1≤n⇒m<n` to the inputs and `m<n⇒m+1≤n` to the outputs? How many lemmas do you want to do this for? Any chance you could provide a list?

Also now I look at those names in #1017 again, I think maybe they'd be better named as `suc[m]≤n⇒m<n` and `m<n⇒suc[m]≤n`?

**Comment by Blaisorblade** [comment]
> Also now I look at those names in #1017 again, I think maybe they'd be better named as suc[m]≤n⇒m<n and m<n⇒suc[m]≤n?

Go ahead :-)

> Hmm so presumably for every lemma of interest, it's just a case of taking the lemma in its standard form, and applying m+1≤n⇒m<n to the inputs and m<n⇒m+1≤n to the outputs?

Uh, interesting idea! I was thinking of taking the existing proofs, which are often more complicated. I suspect it wouldn't work for all those lemmas, but it might be easiest to try.

If the proofs are so easy, it might be less important to include them in the API. Also, depending on plans for #1041 and of the performance of the result, which lemmas are provided might matter less — with sufficient performance, the stdlib might only need enough to bootstrap the Presburger solver.

**Comment by MatthewDaggitt** [comment]
> If the proofs are so easy, it might be less important to include them in the API.

This would also be my feeling.

**Comment by MatthewDaggitt** [comment]
@Blaisorblade unless you have any objections I propose to close this issue? I've just opened a PR #1112 to fix the naming issue, and I can't see any issues using my solution to interface with legacy code.

**Comment by MatthewDaggitt** [comment]
Closed with 368d54f
